### DISCUSSION TITLE
How does "v up" actually work?

### DISCUSSION THREAD

**Discussion by Kostia06** [vlang/v#16119] [General]
I am new to vlang and I found a really cool feature, called "v up", which I don't see in other programming langauges. I want to implement something similar to the "v up" command for my project, in short I want a command that checks if there is a update for my software and if there is update it. Could anyone please help me explain how does the "v up" work?

**Comment by JalonSolov** [comment]
The source is https://github.com/vlang/v/blob/master/cmd/tools/vup.v

**Reply by Kostia06** [reply]
Thank you!

**Reply by Kostia06** [reply]
Could you please explain the process of the source code that you send me?

**Reply by JalonSolov** [reply]
Fairly simple... V does a `git fetch` of the latest commit and checks the hash.  If that hash matches the hash of what it already had, it just prints the message that V is already up-to-date.  If the hash is different, it rebuilds.

**Reply by Kostia06** [reply]
Okay, thank you and one more thing. What does the "v up" actually fetch? The executable or source code?

**Reply by bogen85** [reply]
> Okay, thank you and one more thing. What does the "v up" actually fetch? The executable or source code?

The source code. (from the source repo via git).
See @JalonSolov's reply.

It does a rebuild if the source code changed.

**Reply by Kostia06** [reply]
Oh i see thank you!
